addon/services/sl-behavior.js

```diff
@@ -35,120 +35,70 @@ export default Ember.Service.extend({
     // Methods
 
     /**
-     * Retrieve Behavior Group string from Provider object
+     * Determine whether the desired activity is able to be performed on the resource
      *
      * @function
-     * @param {Object} provider
-     * @throws {ember.assert} If parameter is not an object
-     * @returns {String}
+     * @param {String} activity
+     * @param {String} resource
+     * @param {Boolean} possible
+     * @throws {ember.assert} If no parameters provided, first parameter is not a string, second parameter is not a
+     *     string, or third parameter is not a boolean or undefined
+     * @returns {Boolean}
      */
-    getBehaviorGroup( provider ) {
-        Ember.assert(
-            'services/behavior.getBehaviorGroup() expects a parameter to be provided',
-            provider
-        );
-
-        let behaviorGroupType = Ember.typeOf( provider.behaviorGroup );
-        let behaviorGroupTypeIsString = 'string' === behaviorGroupType;
-        let providerType = Ember.typeOf( provider );
-
+    isAble( activity, resource, possible ) {
         Ember.assert(
-            'services/behavior.getBehaviorGroup() expects parameter to be an Object',
-            'instance' === providerType ||
-            'object' === providerType
+            'services/behavior.isAble() expects at least two parameters to be provided',
+            activity && resource
         );
 
-        Ember.assert(
-            'services/behavior.getBehaviorGroup() `provider.behaviorGroup` must be a String or Function',
-            'function' === behaviorGroupType ||
-            behaviorGroupTypeIsString
-        );
-
-        let behaviorGroup = ( behaviorGroupTypeIsString ) ?
-            Ember.get( provider, 'behaviorGroup' ) :
-            provider.behaviorGroup();
-
-        return behaviorGroup;
-    },
+        const behaviors = this.getWithDefault( 'behaviors', null );
```

Comment:
I agree. The only way `getWithDefault()` would return the default is if the behaviors property were `undefined`. It shouldn't ever be `undefined` if set correctly using `setBehaviors()`. Perhaps an Ember.assert() would be good to put here checking that the type is {Object | null} just in case it is misused somehow?